plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheck.kt

```diff
@@ -0,0 +1,75 @@
+package org.jlleitschuh.gradle.ktlint
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.FileTree
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.Console
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import org.jlleitschuh.gradle.ktlint.reporter.KtlintReport
+import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
+
+open class KtlintCheck : DefaultTask() {
+
+    @get:Classpath
+    val classpath: ConfigurableFileCollection = project.files()
+
+    @get:Internal
+    val sourceDirectories: ConfigurableFileCollection = project.files()
+
+    @get:SkipWhenEmpty
+    @get:PathSensitive(PathSensitivity.RELATIVE)
+    @get:InputFiles
+    val sources: FileTree = sourceDirectories.asFileTree.matching { it.include("**/*.kt") }
+
+    @get:Input
+    val verbose: Property<Boolean> = project.objects.property(Boolean::class.javaObjectType)
```

Comment:
:x: `Boolean::class.java` should be enough

There are many other instances of this below, not commented on each one

Reply:
If I use `Boolean::class.java` I get:
```
Cannot get the value of a property of type boolean as the provider associated with this property returned a value of type java.lang.Boolean.
```
when running the tests.

Reply:
Ah right, fair enough